pkg/netceptor/ping_test.go

```diff
@@ -0,0 +1,416 @@
+package netceptor_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/netceptor"
+	"github.com/ansible/receptor/pkg/netceptor/mock_netceptor"
+	"go.uber.org/mock/gomock"
+)
+
+// Helper function to create test addresses.
+func newTestAddr(node, service, network string) netceptor.Addr {
+	addr := netceptor.Addr{}
+	addr.SetNode(node)
+	addr.SetService(service)
+	addr.SetNetwork(network)
+
+	return addr
+}
+
+func TestNetceptor_Ping(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	tests := []struct {
+		name         string
+		target       string
+		hopsToLive   byte
+		setupMock    func(*netceptor.Netceptor) (*netceptor.Netceptor, error)
+		expectError  bool
+		expectedNode string
+	}{
```

Comment:
remove unused setupMock
```suggestion
	tests := []struct {
		name         string
		target       string
		hopsToLive   byte
		expectError  bool
		expectedNode string
	}{
```